### PR TITLE
Add more log entries on start up

### PR DESCRIPTION
**Pull request by nevalla**
This PR will add more log entries on app start up to get better understanding about start up phases. Also added phase to test `LensProxy` connection by requesting `/version` end point. `WindowManager` is now outputting error if it fails to load Main window url.

```
[0] info: 🚀 Starting Lens from "/Users/lauri/Library/Application Support/LensDev"
[0] info: 📡 Checking for app updates
[0] info: Skip checkForUpdatesAndNotify because application is not packed
[0] info: 🤓 Installing developer tools
[0] info: 💾 Loading stores
[0] info: [STORE]: LOADED from /Users/lauri/Library/Application Support/LensDev/config.json
[0] info: [STORE]: LOADED from /Users/lauri/Library/Application Support/LensDev/lens-cluster-store.json
[0] info: [STORE]: LOADED from /Users/lauri/Library/Application Support/LensDev/lens-workspace-store.json
[0] info: [STORE]: LOADED from /Users/lauri/Library/Application Support/LensDev/lens-extensions.json
[0] info: [STORE]: LOADED from /Users/lauri/Library/Application Support/LensDev/lens-filesystem-provisioner-store.json
[0] info: 🔑 Getting free port for LensProxy server
...
[0] info: 🔌 Starting LensProxy
[0] info: [LENS-PROXY] LensProxy server has started at http://localhost:52418
[0] info: 🔎 Testing LensProxy connection ...
...
[0] info: ⚡ LensProxy connection OK
[0] info: 🖥️  Starting WindowManager
[0] info: 🧩 Initializing extensions
...
```

### CODE REVIEW

**Review by aleksfront — DISMISSED** [review]

**Review by jakolehm — COMMENTED** [review]
Could we also log something about `shellSync()` call?

**Comment by nevalla** [comment]
> Could we also log something about `shellSync()` call?

Yes, definitely.

**Review by jakolehm — DISMISSED** [review]
LGTM

**Comment by nevalla** [comment]
Added log entry about syncing shell environment, PTAL again.

**Review by aleksfront — DISMISSED** [review]

**Comment by nevalla** [comment]
@jakolehm Please take a look again.

File: src/main/index.ts [hunk 1]
```diff
@@ -106,10 +114,33 @@ app.on("ready", async () => {
...
+    const response = await requestPromise({
+      method: "GET",
+      uri: `http://localhost:${proxyPort}/version`,
+      resolveWithFullResponse: true
+    });
+
+    const appVersion = JSON.parse(response.body).version;
+
+    if (getAppVersion() != appVersion) {
+      logger.error(`Proxy server responded with invalid response: ${response.body}`);
```
Comment (Nokel81) [inline src/main/index.ts]: Shouldn't this be `crit`?

File: src/main/index.ts [hunk 2]
```diff
@@ -61,20 +63,24 @@ if (process.env.LENS_DISABLE_GPU) {
...
   });
 
+  logger.info(`📡 Checking for app updates`);
   const updater = new AppUpdater();
 
   updater.start();
 
   registerFileProtocol("static", __static);
 
+  logger.info("🤓 Installing developer tools");
```
Comment (jakolehm) [inline src/main/index.ts]: I think we skip these on production build? Logging this on production is probably wrong?

File: src/main/index.ts [hunk 3]
```diff
@@ -106,10 +114,33 @@ app.on("ready", async () => {
     app.exit();
   }
 
+  // test proxy connection
+  try {
```
Comment (jakolehm) [inline src/main/index.ts:119]: Could we refactor this into a separate function?

File: src/main/index.ts (hunk 2 above)
Comment (jakolehm) [reply src/main/index.ts]: @nevalla ping

Comment (nevalla) [reply src/main/index.ts]: Yes, it's wrong. Moved the log line inside the installation method.

File: src/main/index.ts (hunk 3 above)
Comment (nevalla) [reply src/main/index.ts:119]: Refactored

**Review by jakolehm — DISMISSED** [review]

File: src/main/index.ts (hunk 3 above)
Comment (Nokel81) [reply src/main/index.ts:119]: The try block?

File: src/main/lens-proxy.ts [hunk 4]
```diff
@@ -29,7 +29,7 @@ export class LensProxy {
 
   listen(port = this.port): this {
     this.proxyServer = this.buildCustomProxy().listen(port);
-    logger.info(`LensProxy server has started at ${this.origin}`);
+    logger.info(`[LENS-PROXY] LensProxy server has started at ${this.origin}`);
```
Comment (Nokel81) [inline src/main/lens-proxy.ts]:
```suggestion
    logger.info(`[LENS-PROXY]: Proxy server has started at ${this.origin}`);
```

File: src/main/window-manager.ts [hunk 5]
```diff
@@ -81,10 +82,19 @@ export class WindowManager extends Singleton {
         this.splashWindow = null;
         app.dock?.hide(); // hide icon in dock (mac-os)
       });
+
+      this.mainWindow.webContents.on("did-fail-load", (_event, code, desc) => {
+        logger.error(`[WINDOW-MANAGER] Failed to load Main window`, code, desc);
```
Comment (Nokel81) [inline src/main/window-manager.ts]: Winston does better with objects then raw values.
```suggestion
        logger.error(`[WINDOW-MANAGER]: Failed to load Main window`, { code, desc });
```

File: src/main/window-manager.ts [hunk 6]
```diff
@@ -81,10 +82,19 @@ export class WindowManager extends Singleton {
...
+      });
+
+      this.mainWindow.webContents.on("did-finish-load", () => {
+        logger.info("[WINDOW-MANAGER] Main window loaded");
```
Comment (Nokel81) [inline src/main/window-manager.ts]:
```suggestion
        logger.info("[WINDOW-MANAGER]: Main window loaded");
```

File: src/main/window-manager.ts [hunk 7]
```diff
@@ -81,10 +82,19 @@ export class WindowManager extends Singleton {
...
+      });
     }
 
     try {
       if (showSplash) await this.showSplash();
+      logger.info(`[WINDOW-MANAGER] Loading Main window from url: ${this.mainUrl} ...`);
```
Comment (Nokel81) [inline src/main/window-manager.ts]:
```suggestion
      logger.info(`[WINDOW-MANAGER]: Loading Main window from url: ${this.mainUrl} ...`);
```

File: src/main/index.ts [hunk 8]
```diff
@@ -109,10 +115,27 @@ app.on("ready", async () => {
...
+    }
+    logger.info("⚡ LensProxy connection OK");
+  } catch (error) {
+    logger.error("Checking proxy server connection failed", error);
+  }
+
   extensionLoader.init();
   extensionDiscovery.init();
+
+  logger.info("🖥️  Starting WindowManager");
```
Comment (Nokel81) [inline src/main/index.ts:134]:
```suggestion
  logger.info("🖥️ Starting WindowManager");
```

File: src/main/index.ts (hunk 3 above)
Comment (nevalla) [reply src/main/index.ts:119]: > The try block?

Don't get. What about it?

File: src/main/index.ts (hunk 1 above)
Comment (Nokel81) [reply src/main/index.ts]: NVM, crit doesn't exist

File: src/main/index.ts (hunk 3 above)
Comment (Nokel81) [reply src/main/index.ts:119]: I was expecting the whole try block to be refactored out

Comment (nevalla) [reply src/main/index.ts:119]: Okay... I think it's now in line with other checks present in that file.

Comment (jakolehm) [reply src/main/index.ts:119]: I meant the whole try block but I'm fine with this (it was just a suggestion).

**Review by jakolehm — APPROVED** [review]

**Review by aleksfront — APPROVED** [review]

**Pull request merged**